### ISSUE TITLE
can't get cloud.Bucket name

### ISSUE THREAD

**Issue by ainvoner** [winglang/wing#4907]
### I tried this:

```js
  pub logsBucket: cloud.Bucket;
  init() {
   this.myBucket = new cloud.Bucket();
   let bucketName = myBucket.name;
   
   // creating a cloudFront distribution with the following config:
   loggingConfig: {
     bucket: bucketName,
     includeCookies: true,
     prefix: "reverse-proxy"
   }
 
  ...
  
  }
```

### This happened:

there is no way to get or set the bucket name.

### I expected this:

to have the ability to get the bucket name

### Is there a workaround?

```js
  pub logsBucket: cloud.Bucket;
  init(){
    this.logsBucket = new cloud.Bucket() as "reverse-proxy-logs-bucket";
    let bucket: aws.s3Bucket.S3Bucket = unsafeCast(this.logsBucket).bucket;
    let logsBucketName = bucket.bucket;
  }
```

### Component

SDK

### Wing Version

latest

### Platform(s)

MacOS

### Community Notes

* Please vote by adding a 👍 reaction to the issue to help us prioritize.
* If you are interested to work on this issue, please leave a comment.

**Comment by eladb** [comment]
Ideally a safe downcast is the way to go here. Something like:

```js
bring aws;

let bucket = aws.Bucket.of(cloudBucket);
let bucketName = bucket.bucketName;
```

**Comment by staycoolcall911** [comment]
Closed in #4938
